### PR TITLE
Issue #27606: Minor Bug fixes, mostly renaming

### LINKED ISSUE #27606: Integration with Third Party CA's that consume K8s CSR API
Integrate Third Party CAs (that read/consume K8s CertificateSigningRequests, ex CertManager) with Istio to sign workload certificates.

**Describe the feature request**
The Feature request is detailed in the two design documents below.

https://docs.google.com/document/d/1KAw8-0FivdYQAcWMVTMl-JYvVCn2fpNTbu6Ya3y9d1E/edit#heading=h.22tdfpc7skzk
https://docs.google.com/document/d/1QReA8XjWrWTnErQYPZUdPA5d2O6fc3VkLWr47hY0Nbg/edit#heading=h.xw1gqgyqs5b

Please also refer to the general implementation spec of this feature 
https://docs.google.com/document/d/1X_BSNv7EztNAvND1JJs9YKFzWzn65cYXMeTkPh9ncr4/edit#heading=h.se6wjxq9mtpk

**Describe alternatives you've considered**
Third party CA's can also integrate with the Istio ecosystem by implementing the v1alpha1 CSR API. 
https://github.com/istio/api/blob/master/security/v1alpha1/ca.proto
This effort is being undertaken parallely.

[ ] Docs
[ ] Installation
[ ] Networking
[ ] Performance and Scalability
[ ] Extensions and Telemetry
[X] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

**Additional context**

### PR DESCRIPTION
**Pull request by shankgan**
Please provide a description for what this PR is for.
Minor bug fixes  fix #27606

And to help us figure out who should review this PR, please 
put an X in all the areas that this PR affects.

[ ] Configuration Infrastructure
[ ] Docs
[ ] Installation
[ ] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[X] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

Pull Request Attributes

Please check any characteristics that apply to this pull request. 

[X] Does not have any changes that may affect Istio users.

### CODE REVIEW

File: pilot/pkg/bootstrap/istio_ca.go [hunk 1]
```diff
@@ -44,20 +44,26 @@ import (
...
 	Namespace      string
 	Authenticators []authenticate.Authenticator
 }
 
 const (
-	//KubernatesCA : Kubernates acts as the Certificate Authority
-	extCAK8s string = "KUBERNATES"
-	// extCAGrpc string = "GRPC"
+	// ExtCAK8s : Integrate with external CA using k8s CSR API
+	ExtCAK8s CaExternalType = "ISTIOD_RA_KUBERNETES_API"
```
Comment (Monkeyanator) [inline pilot/pkg/bootstrap/istio_ca.go:62]: "ISTIOD_**C**A_KUBERNETES API"?

Comment (Monkeyanator) [reply pilot/pkg/bootstrap/istio_ca.go:62]: nevermind, just got the context around Registration Authority :)

Comment (shankgan) [reply pilot/pkg/bootstrap/istio_ca.go:62]: Yeah, had a lot of back and forth regarding the nomenclature. This is the one we settled on

**Review by howardjohn — APPROVED** [review]

**Comment by shankgan** [comment]
cherrypick/release-1.8

**Comment by shankgan** [comment]
/cherrypick\/release-1.8

**Review by myidpt — APPROVED** [review]

**Pull request merged**